### PR TITLE
Add support for zabbix services in zabbix module and state.

### PR DESCRIPTION
**Pull request by johnybx**
### What does this PR do?
Adds support for manipulating zabbix services which is useful for calculating SLA and overall availability.  

Example state:
```
create_service_icmp-server-1:
  zabbix_service.present:
    - host: server-1
    - service_root: Server-group/server icmp
    - service_name: server-1
    - trigger_desc: is unavailable by ICMP
```
Result of this state would be (if no services exists yet) that service Server-group would be created with child "server icmp" and than it would have child server-1. Then we may have on this server also apache2 or nginx which should listen on port 443 so we could add service like this:

```
create_service_https-server-1:
  zabbix_service.present:
    - host: server-1
    - service_root: Server-group/server https
    - service_name: server-1
    - trigger_desc: port 443 is closed
```
So this would not create service Server-group but would add service "server https" with child server-1. Leaf services have associated trigger which is identified by name and based on this trigger is calculated availability in zabbix. Also we could just add new server to existing services (as child):
```
create_service_icmp-server-2:
  zabbix_service.present:
    - host: server-2
    - service_root: Server-group/server icmp
    - service_name: server-2
    - trigger_desc: is unavailable by ICMP
```
After this states we would end up with structure like this:

- Server-group
    -  server icmp
        -  server-1 -> associated with trigger
        -  server-2 -> associated with trigger
    -  server https
        -  server-1 -> associated with trigger

There is no limit on how many levels of services we can do. Trigger description in combination with trigger priority (severity) should be unique because for one service there can be only one trigger and first found match will be used.

There are two states present and absent. In module there are functions which create service, get service or delete server.

### CODE REVIEW

File: salt/modules/zabbix.py [hunk 1]
```diff
@@ -2503,3 +2503,187 @@ def configuration_import(config_file, rules=None, file_format='xml', **connectio
...
+def triggerid_get(hostid=None, trigger_desc=None, priority=4, **connection_args):
+    '''
+    .. versionadded:: Fluorine
+
+    Retrieve trigger ID and description based in host ID and trigger description.
+
+    .. note::
+        https://www.zabbix.com/documentation/3.4/manual/api/reference/trigger/get
+
+    :param hostid: ID of the host which's trigger we want to find
```
Comment (cachedout) [inline salt/modules/zabbix.py]: s/which's/whose/

File: salt/modules/zabbix.py [hunk 2]
```diff
@@ -2503,3 +2503,187 @@ def configuration_import(config_file, rules=None, file_format='xml', **connectio
...
+    :param trigger_desc: Description of trigger (trigger name) which's we want to find
```
Comment (cachedout) [inline salt/modules/zabbix.py]: s/which's/whose/

File: salt/modules/zabbix.py [hunk 3]
```diff
@@ -2503,3 +2503,187 @@ def configuration_import(config_file, rules=None, file_format='xml', **connectio
                                                                 'called successfully.'}
     except SaltException as exc:
         return {'name': config_file, 'result': False, 'message': six.text_type(exc)}
+
+
+def triggerid_get(hostid=None, trigger_desc=None, priority=4, **connection_args):
```
Comment (cachedout) [inline salt/modules/zabbix.py]: We typically prefer named kwargs instead of `**connection_args`)

File: salt/modules/zabbix.py [hunk 4]
```diff
@@ -2503,3 +2503,187 @@ def configuration_import(config_file, rules=None, file_format='xml', **connectio
...
+        salt '*' zabbix.triggerid_get 1111 'trigger name to find' 5
+    '''
+    conn_args = _login(**connection_args)
+    ret = False
+    try:
+        if conn_args:
+            method = 'trigger.get'
+            if not hostid or not trigger_desc:
+                return {'result': False, 'comment': 'hostid and trigger_desc params are required'}
+            params = {'output': ["triggerid", "description"],
```
Comment (cachedout) [inline salt/modules/zabbix.py]: Our style guide recommends single quotes here and on the line below.

File: salt/modules/zabbix.py [hunk 5]
```diff
@@ -2503,3 +2503,187 @@ def configuration_import(config_file, rules=None, file_format='xml', **connectio
...
+
+    :return: Service details, False if service could not be added or on failure.
+
+    CLI Example:
+    .. code-block:: bash
+
+        salt '*' zabbix.service_add 11 'My service' 11111
+    '''
+    conn_args = _login(**connection_args)
+    ret = False
```
Comment (cachedout) [inline salt/modules/zabbix.py]: It looks like here and in other places we're returning either a boolean or a dictionary. Should it instead be an empty dict or a popualated dict? That might avoid type errors for consumers of this module.

Comment (johnybx) [reply salt/modules/zabbix.py]: good idea. I changed it in whole module.

File: salt/modules/zabbix.py (hunk 3 above)
Comment (johnybx) [reply salt/modules/zabbix.py]: good idea. I changed it in whole module.

**Comment by johnybx** [comment]
Hi @cachedout, I did requested changes.

File: salt/modules/zabbix.py [hunk 6]
```diff
@@ -427,7 +427,7 @@ def get_object_id_by_params(obj, params=None, **connection_args):
                             'Result: {2}'.format(obj + '.get', params, res))
 
 
-def apiinfo_version(**connection_args):
+def apiinfo_version(**kwargs):
```
Comment (cachedout) [inline salt/modules/zabbix.py]: Could you clarify why this change and the others like it are necessary, please?
